source/server/connection_handler_impl.cc

```diff
@@ -77,15 +77,29 @@ ConnectionHandlerImpl::SslActiveListener::SslActiveListener(
                                                            parent.stats_store_, listener_options),
                      factory, socket.localAddress()->asString()) {}
 
-Network::Listener* ConnectionHandlerImpl::findListenerByPort(uint32_t port) {
-  // findListenerByPort is O(#listeners) operation, may need to add a map<port, listener>
-  // to improve performance to O(log #listeners); since the number of listeners is small
-  // linear performance might be adequate
-  auto l = std::find_if(
-      listeners_.begin(), listeners_.end(),
-      [port](const std::pair<Network::Address::InstancePtr, ActiveListenerPtr>& p) {
-        return p.first->type() == Network::Address::Type::Ip && p.first->ip()->port() == port;
-      });
+Network::Listener*
+ConnectionHandlerImpl::findListenerByAddress(Network::Address::InstancePtr address) {
+  // This is a linear operation, may need to add a map<address, listener> to improve performance.
+  // However, linear performance might be adequate since the number of listeners is small.
+  auto l =
```

Comment:
This was a problem in the original code - we shouldn't use single letter variables like l, too easy to confuse with 1. can we rename to "auto listener" ?